src/utils/deprecationWarning.js

```diff
@@ -25,11 +25,11 @@ function deprecationWarning(oldname, newname, link) {
 
 deprecationWarning.wrapper = (Component, ...args) =>
   class DeprecatedComponent extends Component {
-    componentWillMount(...methodArgs) {
+    UNSAFE_componentWillMount(...methodArgs) {  // eslint-disable-line
```

Comment:
should this just be in the ctor?

Reply:
yeah

Reply:
Let me know if you want me to push a particular change into this pull request.
Otherwise you can pull, make the changes on your branch.

 Looks like there were a couple changes in the pipeline already, hopefully a new v3 release isn't blocked by anything big?